src/background/services/actions/models.ts

```diff
@@ -19,25 +20,32 @@ export type Action<DisplayData = any, Params = any> = JsonRpcRequestPayload<
   DAppProviderRequest | RpcMethod,
   Params
 > & {
+  caipId?: string;
   scope: string;
   context?: Record<string, unknown>;
-  signingData?: SigningData;
   dappInfo?: DappInfo;
   [ACTION_HANDLED_BY_MODULE]?: boolean;
   time?: number;
   status?: ActionStatus;
   result?: any;
   error?: string;
-  displayData: DisplayData;
   // we store the window ID of the confirmation popup so
   // that we can clean up stale actions later
   popupWindowId?: number;
   inAppPromptId?: number;
   actionId?: string;
+} & {
+  signingData?: SigningData;
+  displayData: DisplayData;
+};
+
+export type MultiTxAction = Omit<Action, 'signingData' | 'displayData'> & {
```

Comment:
Are regular transactions a batch of one? Do we need to keep two separate types? Should we just use the one that is capable of handling more and on the UI we just display the single vs batch UI based on the number of transactions needed to be approved?

Reply:
I'm open to doing that as a follow-up PR, as it would involve revamping the current (single) `Action` model on each handler we have, which would blow up this branch tremendously.

To be completely honest, I would like to revisit the `Action` model as a whole and require it to always be typed (for internal request handlers too) -- especially the data we use on the UI side.